tests/db_functions/json/test_json_array_object.py

```diff
@@ -0,0 +1,96 @@
+from django.db.models import F
+from django.db.models.functions import JSONArray, JSONObject
+from django.test import TestCase
+from django.test.testcases import skipUnlessDBFeature
+
+from ..models import Author
+
+
+@skipUnlessDBFeature("has_json_object_function")
+class JSONArrayObjectTests(TestCase):
```

Comment:
I think my personal preference is to put these tests in `tests/db_functions/json/test_json_array.py` as we probably won't plan to make a new test file for each combination of database functions

Reply:
I did this, but because JSONObject requires `has_json_object_function`, but Felix's other suggestion said to use `supports_json_field`, I had to include both conditions in the tests that require both JSONObject and JSONArray. Hopefully that makes sense.